### PR TITLE
Client: Support passive scanning

### PR DESCRIPTION
**Pull request by psiinon**
## Overview
As per the title 😁 

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            HistoryReference hr = extHistory.getHistoryReference(i);
+            if (alertUrl.equals(hr.getURI().toString())) {
+                alert.setHistoryRef(hr);
+                LOGGER.debug("Found href {} for {}" + hr.getHistoryId(), alert.getUri());
+                break;
+            }
+        }
+        if (alert.getHistoryRef() == null) {
+            LOGGER.error(
+                    "Failed to find Href for URL {}, unable to raise alert",
```
Comment (kingthorin) [inline addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java:61]: Here and in comments etc above we should say history reference, vs href which might be confused with URL

File: addOns/client/src/test/java/org/zaproxy/addon/client/pscan/InformationInStorageUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        // Given
+        rule.setEnabled(!enabled);
+        // When
+        rule.setEnabled(enabled);
+        // Then
+        assertThat(rule.isEnabled(), is(enabled));
+    }
+
+    @Test
+    void shouldReturnTheRightRefaults() {
```
Comment (kingthorin) [inline addOns/client/src/test/java/org/zaproxy/addon/client/pscan/InformationInStorageUnitTest.java:65]: Defaults? 😉

File: addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    private ExtensionHistory extHistory;
+
+    public ClientPassiveScanHelper(ExtensionAlert extAlert, ExtensionHistory extHistory) {
+        this.extAlert = extAlert;
+        this.extHistory = extHistory;
+    }
+
+    public void raiseAlert(Alert alert) {
+        // Find suitable Href
+        int lastId = extHistory.getLastHistoryId();
```
[the remark this answers is not in the document]
Comment (psiinon) [reply addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java:43]: I believe so, yes. But I guess we'll see?

**Comment by psiinon** [comment]
Fixed those 😁

File: addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                        "Found history reference {} for {}" + hr.getHistoryId(), alert.getUri());
+                break;
+            }
+        }
+        if (alert.getHistoryRef() == null) {
+            LOGGER.error(
+                    "Failed to find history reference for URL {}, unable to raise alert",
+                    alertUrl,
+                    alert.toPluginXML(alertUrl));
+        } else {
```
Comment (kingthorin) [inline addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java:65]: Extract a URL method?
Extract a method to get the HR or null?

Let this method actually be about raising the alert (or not if no hist ref)...

Comment (psiinon) [reply addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java]: Updated - that look better?

**Review by kingthorin — APPROVED** [review]
LGTM

File: addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java (hunk 4 above)
Comment (kingthorin) [reply addOns/client/src/main/java/org/zaproxy/addon/client/pscan/ClientPassiveScanHelper.java]: I think so 😀

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**